### PR TITLE
Add enforce email for portal logon (default) and Appointment notifications.

### PR DESCRIPTION
**Pull request by sjpadgett**
Forum request https://community.open-emr.org/t/how-to-do-these-modification/11235
Will leave this open to add appointment change notifications to portal feature request from forum
https://community.open-emr.org/t/patient-portal-is-there-a-notification-when-an-appointment-is-made/11244

### CODE REVIEW

**Comment by sjpadgett** [comment]
Okay y'all can quit picking on the portal for a bit...

**Comment by danehrlich1** [comment]
What does this do?

Would this prevent ppl from guessing the username of “Admin” for instance
because it’s like the default in everything?

**Comment by sjpadgett** [comment]
Has nothing to do with username. This allows not requiring an email to log on to portal and sending a notification to provider when an appointment is added/updated so they can review request.
@bradymiller want to check my escaping otherwise am bringing in.

File: library/globals.inc.php [hunk 1]
```diff
@@ -2962,7 +2969,7 @@ function gblTimeZones()
             '0',
             xl('Rx Show State Lic. #')
         ),
-        'rx_show_drug_drug' => array(
+        'rx_show_drug-drug' => array(
```
Comment (bradymiller) [inline library/globals.inc.php]: revert this change in above line

File: portal/add_edit_event_user.php [hunk 2]
```diff
@@ -527,6 +482,15 @@
 
 if ($_POST['form_action'] != "") {
   // Leave
+    $type = $insert ? xlt("A New Appointment") : xlt("An Updated Appointment");
+    $note = $type . " " . xlt("request was received from portal patient ");
+    $note .= text($_SESSION['ptName']) . " " . xlt("regarding appointment dated ") . text($event_date) . " " . text($starttime) . ". ";
+    $note .= !empty($_POST['form_comments']) ? (xlt("Reason ") . text($_POST['form_comments'])) : "";
+    $note .=  ". " . xlt("Use Portal Dashboard to confirm with patient.");
+    $title = xlt("Patient Reminders");
```
Comment (bradymiller) [inline portal/add_edit_event_user.php]: remove all the html escaping from above since you should not be html escaping stuff that goes to database. it will get html escaped after it is pulled from database.
:)

Comment (bradymiller) [reply portal/add_edit_event_user.php]: also shouldn't have any ending whitespace in the constants. pasted below the fixes
```
    $type = $insert ? xl("A New Appointment") : xl("An Updated Appointment");
    $note = $type . " " . xl("request was received from portal patient") . " ";
    $note .= $_SESSION['ptName'] . " " . xl("regarding appointment dated") . " " . $event_date . " " . $starttime . ". ";
    $note .= !empty($_POST['form_comments']) ? (xl("Reason") . " " . $_POST['form_comments']) : "";
    $note .=  ". " . xl("Use Portal Dashboard to confirm with patient.");
    $title = xl("Patient Reminders");
```

Comment (sjpadgett) [reply portal/add_edit_event_user.php]: lol, completely forgot I was writing to database but what would a PR be from me if I wasn't over escaping or not escaping:)

**Comment by sjpadgett** [comment]
okay thanks Brady

**Comment by bradymiller** [comment]
looks good to me
